Geometry/MTDGeometryBuilder/src/MTDGeomUtil.cc

```diff
@@ -0,0 +1,206 @@
+                                               << detId.rawId() << ") is invalid!" << std::dec << std::endl;
+    const ProxyMTDTopology& topoproxy = static_cast<const ProxyMTDTopology&>(thedet->topology());
+    const RectangularMTDTopology& topo = static_cast<const RectangularMTDTopology&>(topoproxy.specificTopology());
+
+    Local3DPoint local_point(topo.localX(row), topo.localY(column), 0.);
+    global_point = thedet->toGlobal(local_point);
+  } else {
+    throw cms::Exception("mtd::MTDGeomUtil") << "detId " << id.rawId() << " not in MTD" << std::endl;
+  }
+  return global_point;
+}
+
+GlobalPoint MTDGeomUtil::getPosition(const DetId& id, const LocalPoint& local_point) const {
+  auto global_point = GlobalPoint(0., 0., 0.);
+  if (isBTL(id)) {
+    BTLDetId detId{id};
+    DetId geoId = detId.geographicalId(MTDTopologyMode::crysLayoutFromTopoMode(getTopology()->getMTDTopologyMode()));
+    const MTDGeomDet* thedet = geom_->idToDet(geoId);
+    if (thedet == nullptr)
+      throw cms::Exception("mtd::MTDGeomUtil") << "GeographicalID: " << std::hex << geoId.rawId() << " ("
+                                               << detId.rawId() << ") is invalid!" << std::dec << std::endl;
+    const ProxyMTDTopology& topoproxy = static_cast<const ProxyMTDTopology&>(thedet->topology());
+    const RectangularMTDTopology& topo = static_cast<const RectangularMTDTopology&>(topoproxy.specificTopology());
+    auto local_point_sim =
+        topo.pixelToModuleLocalPoint(local_point, detId.row(topo.nrows()), detId.column(topo.nrows()));
+    global_point = thedet->toGlobal(local_point_sim);
+  } else if (isETL(id)) {
+    ETLDetId detId{id};
+    DetId geoId = detId.geographicalId();
+    const MTDGeomDet* thedet = geom_->idToDet(geoId);
+    if (thedet == nullptr)
+      throw cms::Exception("mtd::MTDGeomUtil") << "GeographicalID: " << std::hex << geoId.rawId() << " ("
+                                               << detId.rawId() << ") is invalid!" << std::dec << std::endl;
+    global_point = thedet->toGlobal(local_point);
+  } else {
+    throw cms::Exception("mtd::MTDGeomUtil") << "detId " << id.rawId() << " not in MTD" << std::endl;
+  }
+  return global_point;
+}
+
+int MTDGeomUtil::zside(const DetId& id) const {
+  const MTDDetId hid(id);
+  return hid.zside();
+}
+
+unsigned int MTDGeomUtil::getLayer(const DetId& id) const {
+  unsigned int layer(0);
+  if (isETL(id)) {
+    ETLDetId hid(id);
+    layer = hid.nDisc();
+  } else {
+    layer = 0;
+  }
+  return layer;
+}
+
+int MTDGeomUtil::getModule(const DetId& id) const {
+  int module = -1;
+  if (isETL(id)) {
+    ETLDetId hid(id);
+    module = hid.module();
+  } else {
+    BTLDetId hid(id);
+    module = hid.module();
+  }
+  return module;
+}
+
+// returns the local position as a pair (x, y) - for ETL
+std::pair<float, float> MTDGeomUtil::getPixelInModule(const DetId& id, const int row, const int column) const {
+  ETLDetId detId(id);
```

Comment:
@AuroraPerego why is only the ETL case considered in this method, while also BTL is treated in the following one?

Reply:
because row and column are needed only for ETL to obtain the pixel position, while for BTL they are retrieved from the topology. So there is no need to use this method for BTL

Reply:
in principle you are not interested to do it, but the method should be better capable to handle any possible case, or have a protection preventing a BTL id being passed